### PR TITLE
fix client admins not being able to see all Proposals they are involved with

### PR DESCRIPTION
**Pull request by afeld**
Issue introduced in #538.

More importantly, converted the queries to Arel, which makes the various conditions for the subsets of Proposals composable. http://www.scuttle.io/ was super helpful in doing the conversion. Want to play around with them more to try and make them more readable, but I think they're ok for now.

Both pull requests are in service of https://www.pivotaltracker.com/story/show/99002900.

### CODE REVIEW

File: app/models/approval.rb [hunk 1]
```diff
@@ -9,6 +9,9 @@ class Approval < ActiveRecord::Base
   end
 
   belongs_to :proposal
+  belongs_to :user
+  has_many :delegations, through: :user, source: :outgoing_delegates
+  has_many :delegates, through: :delegations, source: :assignee
```
Comment (afeld) [inline app/models/approval.rb]: Didn't end up using this, but I think it's useful anyway.

Comment (cmc333333) [inline app/models/approval.rb]: Should these be on `Approvals::Individual` instead?

**Comment by afeld** [comment]
Updated!

**Comment by cmc333333** [comment]
I had issues reading this (I followed the method names like I'd follow GOTOs); there is probably room for improvement in the method names, though I don't have a specific suggestion. That said, I think it's a step in the right direction. We'll be able compose these Arel clauses with client-specific clauses to create reports, alerts, etc. relatively easily.

**Comment by afeld** [comment]
> there is probably room for improvement in the method names

:+1:

**Pull request merged**